docs/config.md

```diff
@@ -0,0 +1,64 @@
+# Terramate Configuration
+
+There is a series of different configurations that can be done
+on Terramate, ranging from avoiding duplication to controlling
+order of execution of stacks.
+
+In order to do so Terramate works with configuration files named
+**terramate.tm.hcl**. These files can be found on any directory
+of a Terramate project, but some configurations will be defined
+only once for an entire project, while others provide merge/overriding
+strategies, while others control only behavior specific for
+stacks.
+
+Here is defined the different kinds of configurations and
+how you can learn more about them. But before getting into the different
+kinds of configurations, what would be a Terramate project ?
+
+A Terramate project is essentially any collection of terraform code
+organized into stacks. You can have all terraform code together in a single
+directory but that would defeat the purpose of most Terramate features, so
+usually you have a set of stacks and maybe a set of terraform modules if they
+are not maintained in a different repository.
+
+It is not a hard requirement for Terramate to work that the project uses git 
+for version control, but features like change detection do depend on git to
+work and will fail if this requirement is not met.
+
+So in general, a Terramate project looks like this:
+
+* A git project
+* The git top level dir is the project base dir
+* Stacks will be organized as different directories
+* Configuration may be present on any dir
+
+# Project Configuration
+
+Per project configuration can be defined only once at the project base dir.
+
+Available project wide configurations can be found [here](project-config.md)
+
+# Stack Configuration
+
+Any dir that has no child dirs (a leaf in the project hierarchy) and
```

Comment:
is this a requirement for the new design? Can't we use the `stack {}` block for the definition of a stack? It seems odd that a stack needs to be a leaf, this would conflict with people's way of organizing IaC.

Reply:
This is being discussed here: https://app.shortcut.com/mineiros/story/4218/spec-create-spec-for-hierarchical-nature-of-config-files#activity-4367

Quite open to changing the spec if we agree on using stack{}

Reply:
Updated the spec according to what was discussed here: https://mineirosio.slack.com/archives/C02GFAAAKPT/p1639504351015800